master/buildbot/db/buildrequests.py

```diff
@@ -220,45 +313,107 @@ def getResults(query):
         return self.db.pool.do(thd)
 
     @with_master_objectid
-    def getOldestBuildRequestInQueue(self, buildername, _master_objectid=None):
+    def getPrioritizedBuildRequestsInQueue(self, queue, buildername=None, sourcestamps=None,
```

Comment:
Can you add some documentation to clear up what this is used for.

Also is there any time when you don't want the order? Maybe we don't need to offer the option?